SingularityService/src/main/java/com/hubspot/singularity/data/AbstractMachineManager.java

```diff
@@ -203,20 +222,33 @@ private String getHistoryUpdatePath(SingularityMachineStateHistoryUpdate history
     return ZKPaths.makePath(getHistoryPath(historyUpdate.getObjectId()), historyChildPath);
   }
 
+  public void clearOldHistory(String machineId) {
+    List<SingularityMachineStateHistoryUpdate> histories = getHistory(machineId);
+    histories.sort(Comparator.comparingLong(SingularityMachineStateHistoryUpdate::getTimestamp));
```

Comment:
I think we want to reverse this sort, or we'll keep the oldest `n` entries instead of the newest.

Reply:
Reversing the sort. Though I'm realizing we don't even really expose this data anywhere... Maybe don't even need it